purchase_procurement_run_buy_hook/hooks.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, _
+from odoo.exceptions import UserError
+from odoo.addons.purchase.models.purchase import ProcurementRule
+
+
+def post_load_hook():
+
+    @api.multi
+    def _new_run_buy(self, product_id, product_qty, product_uom,
+                     location_id, name, origin, values):
+        if not hasattr(self, '_prepare_purchase_order_line_update'):
+            return self.action_invoice_create_original(
```

Comment:
Where is this declared? I am not able to find it

Reply:
Ask @rgarnau

Reply:
@lreficent you're right. I was calling a wrong method. I now changed to call '_original_run_buy'.